datafusion/expr/src/type_coercion/binary.rs

```diff
@@ -439,6 +448,12 @@ fn both_numeric_or_null_and_numeric(lhs_type: &DataType, rhs_type: &DataType) ->
     match (lhs_type, rhs_type) {
         (_, DataType::Null) => is_numeric(lhs_type),
         (DataType::Null, _) => is_numeric(rhs_type),
+        (DataType::Dictionary(_, value_type), _) => {
+            is_numeric(value_type) && is_numeric(rhs_type)
+        }
+        (_, DataType::Dictionary(_, value_type)) => {
```

Comment:
I wonder if this need to handle the case when both lhs and rhs are Dictionaries 🤔

Reply:
Actually I'm wondering what Dictionary for `ScalaValue`(rhs here) means although I have seen DataFusion has it.

It sounds reasonable to add the following:

```
(DataType::Dictionary(_, lhs_value_type), DataType::Dictionary(_, rhs_value_type2)) => {
  is_numeric(lhs_value_type) && is_numeric(rhs_value_type2)
}
```